clang/include/clang/Driver/Options.td

```diff
@@ -4467,6 +4467,9 @@ def fintrinsic_modules_path : Separate<["-"], "fintrinsic-modules-path">,  Group
   HelpText<"Specify where to find the compiled intrinsic modules">,
   DocBrief<[{This option specifies the location of pre-compiled intrinsic modules, 
   if they are not in the default location expected by the compiler.}]>;
+
+def emit_fir : Flag<["-"], "emit-fir">, Group<Action_Group>,
+  HelpText<"Build the parse tree, then convert to FIR and dump">;
```

Comment:
IMO, `-emit-mlir` would be a bit more accurate. IIUC, FIR is just one of (potentially) many dialects that the generated output will contain, right? But I'm new to this, so might be missing something key. 

I used `-emit-fir` as that's what `bbc` implements. We should also decide what to do in `bbc`. Perhaps allow both `-emit-mlir` and `-emit-fir`? 

@schweitzpgi , any thoughts? Perhaps I should ping flang-dev as well.

Reply:
I don't really have a strong opinion on the spelling of this option in the flang driver.

Once upon a time, it was pretty trivial to add alias option spelling in LLVM. I don't know if that is still possible or not, but it would be a way to "support both".

Reply:
I have a slight preference for `-emit-mlir` as the more accurate alternative and have changed the spelling in the latest patch.

Yes, we could use aliases here. I also pinged [flang-dev](https://lists.llvm.org/pipermail/flang-dev/2021-September/000792.html) for better visibility.